pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -50,15 +149,25 @@ func (lp *LifecycleProcessor) OnInvokeEnd(endDetails *InvocationEndDetails) {
 	log.Debugf("[lifecycle] Invocation isError is: %v", endDetails.IsError)
 	log.Debug("[lifecycle] ---------------------------------------")
 
+	statusCode, err := trigger.GetStatusCodeFromHTTPResponse([]byte(parseLambdaPayload(endDetails.ResponseRawPayload)))
+	if err != nil {
+		log.Debugf("[lifecycle] Couldn't parse response payload: %v", err)
+	}
+
+	// This will only add the status code if it comes from an HTTP-like
+	// response struct
+	lp.addTag("http.status_code", statusCode)
+
 	if !lp.DetectLambdaLibrary() {
 		log.Debug("Creating and sending function execution span for invocation")
-		endExecutionSpan(lp.ProcessTrace, endDetails.RequestID, endDetails.EndTime, endDetails.IsError, endDetails.ResponseRawPayload)
+		endExecutionSpan(lp.GetExecutionInfo(), lp.requestHandler.triggerTags, lp.ProcessTrace, endDetails.RequestID, endDetails.EndTime, endDetails.IsError, endDetails.ResponseRawPayload)
```

Comment:
We don't have to change this right now, but this is a lot of arguments for this function. An alternative would be to pass a struct with these fields. A struct is more explicit, and is easier to propagate down a callstack.

Reply:
I think this and startExecutionSpan should both be refactored eventually because they're both kind of monolithic functions which each do numerous things. Ideally I'd refactor this, but I think this is okay for now just so we have something that works.